src/js/jsx/shared/Datalist.jsx

```diff
@@ -451,6 +475,15 @@ define(function (require, exports, module) {
             }
         },
 
+        removeAutofillSuggestion: function () {
+            if (this.props.useAutofill) {
+                this.setState({
+                    suggestTitle: ""
+                });
+                this.forceUpdate();
```

Comment:
Yeah this is bad... SCU should not ever update anything. The render method should see the conditions you're checking for in SCU and render correctly if the condition is being met.